model/event/event_logger.go

```diff
@@ -24,26 +25,50 @@ func NewDBEventLogger(collection string) *DBEventLogger {
 	}
 }
 
-func (l *DBEventLogger) LogEvent(event *EventLogEntry) error {
-	if event.Data == nil {
+func (l *DBEventLogger) LogEvent(e *EventLogEntry) error {
+	if err := e.validateEvent(); err != nil {
+		return errors.Wrap(err, "not logging event, event is invalid")
+	}
+	return db.Insert(l.collection, e)
+}
+
+func (l *DBEventLogger) LogManyEvents(events []EventLogEntry) error {
+	catcher := grip.NewBasicCatcher()
+	for i := range events {
+		e := &events[i]
+		if err := e.validateEvent(); err != nil {
+			catcher.Add(err)
+		}
+	}
+	if catcher.HasErrors() {
+		return errors.Errorf("not logging events, some events are invalid: %s", catcher.String())
+	}
+	interfaces := make([]interface{}, len(events))
+	for i := range events {
+		interfaces[i] = &events[i]
+	}
```

Comment:
why not combine these loops so we only iterate over things once

Reply:
could also use a carefully configured buffered inserter from anser to avoid needing to do the type conversion?